pkg/pillar/utils/cloudconfig/cloudconfig.go

```diff
@@ -0,0 +1,88 @@
+package cloudconfig
+
+import (
+	"encoding/base64"
+	"errors"
+	"fmt"
+	"os"
+	"path/filepath"
+	"strconv"
+	"strings"
+
+	"github.com/lf-edge/eve/pkg/pillar/base"
+	fileutils "github.com/lf-edge/eve/pkg/pillar/utils/file"
+	"gopkg.in/yaml.v2"
+)
+
+type CloudConfig struct {
+	RunCmd     []string       `yaml:"runcmd"`
+	WriteFiles []WritableFile `yaml:"write_files"`
+}
+
+type WritableFile struct {
+	Path        string `yaml:"path"`
+	Content     string `yaml:"content"`
```

Comment:
Should we document and enforce any size limits on content?
FWIW I think the controller(s) might have a large limit like 1MB, but maybe we want something smaller here.

Reply:
The only information that I could find on the topic is that some cloud providers (e.g. AWS) restrict the size to [16kB](https://cloudinit.readthedocs.io/en/latest/explanation/format.html#gzip-compressed-content). Since this restriction is done on the cloud side I think that our controller imposed limit of 1MB might be sufficient.

The only concern that I see here is that the files can fill up EVE's storage, but that should be a decisive factor given that container's sizes are usually in tens of MBs

Reply:
Ok